components/browser/icons/src/main/java/mozilla/components/browser/icons/utils/DiskCache.kt

```diff
@@ -61,11 +61,11 @@ internal class DiskCache :
         return emptyList()
     }
 
-    internal fun putResources(context: Context, request: IconRequest) {
+    internal fun putResources(cacheDir: File, request: IconRequest) {
```

Comment:
As mentioned in the other PR: I do not think I agree with such a refactoring: `Context` is needed to lookup a  directory. Passing in a `File` is way more generic and requires the caller to know which File to pass in (which is meant to be an implementation detail). This doesn't protect against a caller providing wrong/other directories.

Reply:
For private/internal functions, is that an issue?

Reply:
I agree that it's less an issue than a public API - but I expect many people to work on that code over time... :)